container/container_unix.go

```diff
@@ -343,6 +343,17 @@ func (container *Container) BuildCreateEndpointOptions(n libnetwork.Network) ([]
 		createOptions = append(createOptions, libnetwork.EndpointOptionGeneric(genericOption))
 	}
 
+	/* FIXME: When restoring, the container should get the same IP address it had before, so
```

Comment:
Yes, the commented code will not achieve the intended thing, even with the new changes in libnetwork.

Regardless the fact that the preferred IP option will be added to docker run and docker network connect UI soon, on restore we do not think it makes much sense to try to restore the same IP address for the container.
The IP container was using may not be available at the time the container is restored.

IP is disposable mutable entity for the container, as long as L3 reachability is in place and name to address resolution is up to date, apps should not worry about which IP the container's interface are using.